src/classes/UTIL_OrgTelemetry_BATCH.cls

```diff
@@ -108,4 +117,23 @@ public without sharing class UTIL_OrgTelemetry_BATCH implements Database.Batchab
             System.FeatureManagement.setPackageIntegerValue(UTIL_OrgTelemetry_SVC.TelemetryParameterName.Data_CountRdsWithDiffAmount.name(), 0);
         }
     }
+
+    /**
+     * @description Batch job to count the number of Data Imports.
+     */
+    private void handleDataImportCounts() {
+        Integer cnt = Database.countQuery('SELECT Count() FROM DataImport__c LIMIT 1');
+        if (cnt == 1) {
+            Database.executeBatch(
```

Comment:
@psadie If you're just counting a small number of records (i.e., DataImport records created in the last 30 days) there probably isn't a need to use a batch job. You can just issue the count() query in this method. The batch job is used to handle counts against really large objects like Opportunity or Recurring Donation. Not that the batch job doesn't work, but it's not always necessary. For example, we're not using the batch job to count error rows, even when there could be millions of error row records in the org.

Reply:
Ah that makes a lot of sense. I'll make that update. Thank you!

Reply:
@force2b - Could I do something like below in UTIL_OrgTelemetrySVC and potentially not have to touch any of batch related classes?

```
/**
* @description Total number of BDI and BGE records created in the last 30 days
*/
private void handleDataImportCount() {
    try {
        Integer bdiCount = Database.countQuery('SELECT count() FROM DataImport__c WHERE CreatedDate = Last_n_days:30');
        featuremanager.setPackageIntegerValue(TelemetryParameterName.Data_CountDataImportsBDILast30Days.name(), bdiCount);

        Integer bgeCount = Database.countQuery('SELECT count() FROM DataImport__c WHERE CreatedDate = Last_n_days:30 AND NPSP_Data_Import_Batch__r.GiftBatch__c = true');
        featuremanager.setPackageIntegerValue(TelemetryParameterName.Data_CountDataImportsBGELast30Days.name(), bgeCount);
    } catch (Exception e) {
        // Default these to -1 so we can determine (in the LMO) if there was an error retrieving them
        featuremanager.setPackageIntegerValue(TelemetryParameterName.Data_CountDataImportsBDILast30Days.name(), -1);
        featuremanager.setPackageIntegerValue(TelemetryParameterName.Data_CountDataImportsBGELast30Days.name(), -1);
    }
}
```

Reply:
Absolutely. My only comment is that this assumes both count queries can be completed in a single execution context. That may be a perfectly fine assumption given expected row counts in those objects. However if you feel that issuing both count queries back to back in the same context could run a little long this could be split into two methods (which is how you've split up the categories anyway - `Data_CountBDIRows` and `Data_CountBGERows` - in the `processTelemetryType()` method.